bba/bba_internal_test.go

```diff
@@ -0,0 +1,435 @@
+				t.Fatalf("expected bval is %t, but got %t", receivedBval.Value, one)
+			}
+		}
+	})
+	tester.setupAssert(7, func(t *testing.T, bbaInstance *BBA) {
+		if !bbaInstance.broadcastedBvalSet.exist(zero) {
+			t.Fatalf("broadcasted bval set not include f+1 sent binary value")
+		}
+		//
+		// test broadcasted message
+		//
+		if len(broadcaster.BroadcastedMessageList) != 2*len(bvalList) {
+			t.Fatalf("expected broadcasted message size is %d, but got %d", 2*len(bvalList), len(broadcaster.BroadcastedMessageList))
+		}
+		for _, msg := range broadcaster.BroadcastedMessageList[len(bvalList):] {
+			req, ok := msg.Payload.(*pb.Message_Bba)
+			if !ok {
+				t.Fatalf("expected payload type is %+v, but got %+v", pb.Message_Bba{}, req)
+			}
+			receivedBval := &bvalRequest{}
+			if err := json.Unmarshal(req.Bba.Payload, receivedBval); err != nil {
+				t.Fatalf("unmarshal bval request failed with error: %s", err.Error())
+			}
+			if receivedBval.Value != zero {
+				t.Fatalf("expected bval is %t, but got %t", receivedBval.Value, zero)
+			}
+		}
+	})
+	tester.setupAssert(9, func(t *testing.T, bbaInstance *BBA) {
+		if bbaInstance.binValueSet.exist(one) {
+			t.Fatalf("binValue set **include** f+1 sent binary value")
+		}
+		if bbaInstance.binValueSet.exist(zero) {
+			t.Fatalf("binValue set **include** f+1 sent binary value")
+		}
+	})
+
+	for i, bval := range bvalList {
+		sender := cleisthenes.Member{
+			Address: cleisthenes.Address{
+				Ip:   "localhost" + strconv.Itoa(i),
+				Port: 8080,
+			},
+		}
+		if err := bbaInstance.handleBvalRequest(sender, bval); err != nil {
+			t.Fatalf("handle bval request failed with error: %s", err.Error())
+		}
+
+		if f, ok := tester.assert(i); ok {
+			f(t, bbaInstance)
+		}
+	}
+	done <- struct{}{}
+}
+
+func handleAuxRequestTestSetup(t *testing.T, auxList []*auxRequest) (*BBA, *mock.Broadcaster, *bbaTester, func()) {
+	tester := newBbaTester()
+	reqList, err := tester.setupAuxRequestList(auxList)
+	if err != nil {
+		t.Fatalf("failed to setup request list: err=%s", err)
+	}
+	bvalRepo := tester.setupRequestRepository()
+	auxRepo := tester.setupRequestRepository()
+	broadcaster := tester.setupBroadcaster(reqList)
+
+	bbaInstance := &BBA{
+		n:                   10,
+		f:                   3,
+		bvalRepo:            bvalRepo,
+		auxRepo:             auxRepo,
+		tryoutAgreementChan: make(chan struct{}, 1),
+		binValueSet:         newBinarySet(),
+		broadcastedBvalSet:  newBinarySet(),
+		broadcaster:         broadcaster,
+	}
+	return bbaInstance, broadcaster, tester, func() {
+		close(bbaInstance.tryoutAgreementChan)
+	}
+}
+
+func TestBBA_HandleAuxRequest(t *testing.T) {
+	auxList := []*auxRequest{
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+		{Value: one},
+	}
+
+	bbaInstance, _, tester, teardown := handleAuxRequestTestSetup(t, auxList)
+	defer teardown()
+
+	tester.setupAssert(6, func(t *testing.T, bbaInstance *BBA) {
```

Comment:
it looks better to explain as comment : because of index start from 0, it is one less than `len`